### ISSUE TITLE
Has the maintainer abandoned this project?

### ISSUE THREAD

**Issue by elig0n** [kvakulo/Switcheroo#117]
As Regin Larsen (@kvakulo ) didn't reply to my e-mail messages for quite some time and as the last commit was 3 months ago and 12 PR of mine are still being held I am guessing that the project maintainer might have abandoned the maintaining of this project. 

This is an excellent project with a lot of potential so I hope that the role of maintaining this project can move on to someone who is interested in keeping this project alive.

**Comment by railapex** [comment]
Hey @elig0n, I see you and @daanzu and some others have a lot of PR's out and nobody's merged them.

Has anybody though about forking this off, combining all of the PR's, and coming up with a new release? Been a year since the current maintainer did a release. Or is there already a combined version somewhere else that I can't find?

**Comment by elig0n** [comment]
I don't think there is, @programcsharp . Nonetheless I have opened a Discord channel to further discuss it - anyone interested is invited to join https://discord.gg/8zMj7fz

**Comment by elig0n** [comment]
@talha131 @programcsharp You can check my fork as it has branches with more features that I implemented.  If you think some are worth may then open an Issue with the ones you think are worth getting into a release and I will consider incorporating them into a version. The more people will show their interest the more chance I will incorporate them.

**Comment by talha131** [comment]
@elig0n Do you have any latest release here?

https://github.com/elig0n/Switcheroo

The last release was a year ago.

**Comment by elig0n** [comment]
@talha131 No. I do not do "latest" releases as I am not the owner of this project. Some of my commits are still pending in here, while my fork even has some more feature implementations. If you know how to build a project with Visual Studio you may pick the ones you like and incorporate & build them yourself. That said I might release a master with changes I decide are worth it to enter a build but still it will not be main Switcheroo line release.

**Comment by talha131** [comment]
> That said I might release a master with changes I decide are worth it to enter a build but still it will not be main Switcheroo line release.

That's fine. I understand that a release from you will not be a main Switcheroo line release. 

You have been following this project longer than me. If you do a release from your fork it would be easier for people like me to get the fixes.

**Comment by elig0n** [comment]
@talha131 So far I didn't do any bug fixes but mainly features for convenience and customization. 
I have merged my commits into the master. You'll have to build it yourself though for the current time being

**Comment by Wizek** [comment]
Hello @daanzu and @elig0n!

Would one/both of you by chance be open to enabling GitHub issues on your respective fork repositories? Or someone else who have forked? That might help reviving this project more. From the [GH network graph](https://github.com/kvakulo/Switcheroo/network) the two of you seem more active maintainers currently AFAICS.

It seems doubtful at this point that @kvakulo will revive this mainline repo, therefore while issues can be raised here and we can discuss them here, they won't be able to be closed or cleaned up.

Also, I took a crack at building your master branch @daanzu, because the slowdown when showing the window (https://github.com/kvakulo/Switcheroo/pull/118) was quite bad for me too:

[switcheroo-daanzu.zip](https://github.com/kvakulo/Switcheroo/files/4632710/switcheroo-daanzu.zip)

I also wouldn't mind binary releases, though building wasn't as bad as I feared. Maybe Travis CI can help build release binaries in a free, automated and simple way?

**Comment by elig0n** [comment]
@Wizek Issues are already enabled on [my fork](https://github.com/elig0n/Switcheroo/issues). Feel free to use them.
Did @daanzu speed improving patches showed noticeable improvement for you ? 
Let me know as I might soon enough merge them. You can also comment [in the relevant pull request page](https://github.com/elig0n/Switcheroo/pull/14).

**Comment by Wizek** [comment]
> @Wizek Issues are already enabled on my fork. Feel free to use them.

Wonderful! I somehow missed it that your repo already has issues enabled! Great! So I guess yours is now the unofficially official current mainline; congratulations! :)

> Did @daanzu speed improving patches showed noticeable improvement for you ?

Gonna reply to this [over there](https://github.com/elig0n/Switcheroo/pull/14)!

**Comment by daanzu** [comment]
@Wizek I guess I am too slow 😉 , but I went ahead and posted what I have been currently using myself, and enabled issues on my fork as well at https://github.com/daanzu/Switcheroo.

**Comment by Wizek** [comment]
Great, @daanzu! In that case I am not quite sure how to decide where the main repo should live. Ideally there would be one canonical version where all the useful features are merged into. While if we make any one of you "the main maintainer", that person might also vanish, (just like it happened twice already in this project's past IIRC) so having two of you could likely be better? Perhaps one (or both!) of you could add the other to their repo as contributor with push access? So we can maybe do something like the following article describes? https://felixge.de/2013/03/11/the-pull-request-hack.html

**Comment by Wizek** [comment]
Ooooh! And I'm just now seeing that both of you have made binary releases with version tags, I'm really glad for those too! :) I wonder how divergent the two forks are right now and whether we can create a unified variant! :)

**Comment by daanzu** [comment]
@Wizek FWIW, I don't have any big plans for my fork. I just made some minor fixes that I thought were nice to have, and have been using it happily tens or hundreds of times a day.

**Comment by Wizek** [comment]
I do have a few wishes which if not very hard I'd like to see in one of your versions, @daanzu or @elig0n! :)  

1. While the speedup is very nice, it still takes 1-2 seconds sometimes after I press Alt+Space. If further speedup, or my [background-cache-refresh idea](https://github.com/elig0n/Switcheroo/pull/14#issuecomment-629781055) is unwieldy to implement, could we at least show the window instantly and say "loading..." in the place of the list? This way I am sometimes left wondering if I even pressed the key combination at all. 

2. Dark/night mode? Windows 10 has this built in. I use dark mode at night, and Switcheroo can be a bit blinding. (Somewhat can be alleviated by solving no. 3.) Automatic adherence to this could be ideal and/or a simple radio button setting in options:

    ```
    Theme:     ( ) Light     ( ) Dark     (o) System
    ```

3. Set a limit to height of the popup, either hard-coded or configurable in options? My monitor is 1440 pixels high, it's a bit overwhelming when all that height is used, especially to show a dense list of text of about 50 items that I definitely can't take in at a glance. E.g. only using 60-90% of max height could be ideal. Note how neither does the Alt+Tab popup fill the whole screen as to not be overwhelming. I can always scroll if I want to, and I rarely do; instead I pick from the top of the list or start typing to search. 

Now, if (some of) these are easy to implement and you are happy to, then I'm all the more glad! But if not, I'm also happy to perhaps take some pointers as to where in the codebase I should start poking around. How would you feel about giving me some pointers? And even though I don't have a lot of experience with C#, maybe I can fumble around till I can put together a PR. At any rate it would do me good if I learnt [alternative and potentially much more reliable ways of responding to global keyboard shortcuts in Windows](https://github.com/wilix-team/iohook/issues/93). Which I could perhaps do if I familiarize myself with the internals of this project.

**Comment by elig0n** [comment]
@Wizek I have recently merged dark-light theme from @crar01 to my master please please make sure to check my release commits and/or readme page for any new changes and incorporations before asking them . 

Your other suggestions will be positively considered as well..in general I wish to introduce further customizability in the future. 

Feel free to open further features requests at the Issues page on my repo or come for a discussion on the discord channel, it will be more convenient and functional for me to get back to your requests there than in this thread.
